subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -144,10 +144,8 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec {
         when:
         run 'consumer'
         then:
-        // FIXME: Should have been executed
-        notExecuted(':generator')
-        // FIXME: Should have been executed
-        skipped(':consumer')
+        executedAndNotSkipped(':generator')
+        executedAndNotSkipped(':consumer')
```

Comment:
Now that the Provider comes out of our `ValidatingValue`, the dependency can actually be discovered and is not hidden anymore by unpacking the Provider as part of obtaining the value.

Reply:
Because we don't unpack the value but return the `Provider` instance, right?

Reply:
Exactly.

Reply:
I marked #3811 with milestone 4.7 RC1.

Reply:
Isn't this a little different from #3811? In one example, they're using `newOutputFile()`, but in the other, they're using `layout.fileProperty()` for the _output_ property.

In this test, we're only using `newOutputFile()`.  I think this was fixed because of the changes to  handle providers used with `@Nested`?

I don't think we have to make `layout.fileProperty()` work in the same way that `newOutputFile()` does.

Reply:
OK, so this PR makes using `newInputFile()` unnecessary. It is enough to annotate the input file property with `@InputFile` and to use `layout.fileProperty()`.

This PR doesn't make `newOutputFile()` unnecessary, with is also part of #3811.

Do you think the asymmetry makes sense? To me, it seems like the same output property cannot be shared between tasks (only one task can produce the file), while an input property (`layout.fileProperty()`, not `newInputFile()`) can be shared via a `@Nested` bean. WDYT?

Reply:
What's our plan with `newInputFile()` and `newOutputFile()`?